projects/smoke_bts7200/inc/current_measurement.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+// Periodically reads current from load switches and exposes global storage.
+// Requires GPIO, interrupts, soft timers, ADC (in ADC_MODE_SINGLE), and I2C to be initalized.
+
+#include <stdint.h>
+#include "currents.h"
+#include "gpio.h"
+#include "mux.h"
+#include "pca9539r_gpio_expander.h"
+#include "status.h"
+
+#define MAX_POWER_DISTRIBUTION_BTS7200_CHANNELS 16  // max BTS7200s per board
+#define MAX_POWER_DISTRIBUTION_BTS7040_CHANNELS 16  // max BTS7040s per board
+
+typedef void (*PowerDistributionCurrentMeasurementCallback)(void *context);
+
+typedef struct {
+  Pca9539rGpioAddress dsel_pin;
+  PowerDistributionCurrent current_0;
+  PowerDistributionCurrent current_1;
+  uint8_t mux_selection;
+} PowerDistributionBts7200Data;
+
+typedef struct {
+  PowerDistributionCurrent current;
+  uint8_t mux_selection;
+} PowerDistributionBts7040Data;
+
+typedef struct {
+  I2CPort i2c_port;
+
+  // All distinct I2C addresses on which there are BTS7200 DSEL pins
+  I2CAddress *dsel_i2c_addresses;
+  uint8_t num_dsel_i2c_addresses;  // length of preceding array
+
+  // Data of all BTS7200s from which to read
+  PowerDistributionBts7200Data *bts7200s;
+  uint8_t num_bts7200_channels;  // length of preceding array
+
+  // Data of all BTS7040/7008s from which to read
+  PowerDistributionBts7040Data *bts7040s;
+  uint8_t num_bts7040_channels;  // length of preceding array
+
+  MuxAddress mux_address;
+} PowerDistributionCurrentHardwareConfig;
```

Comment:
This is the only section of this header file you actually need, all the rest isn't used in the smoke test.

Reply:
You could also probably just move this struct to `current_measurement_config.h` and delete this whole file.